---
layout: page
title: Microbiome data analysis tutorials and workflows
---

* [Metagenome prediction with PICRUSt2 tutorial](https://github.com/picrust/picrust2/wiki) - Check linked wiki page for latest version
   
* [Resources for: Canadian Bioinformatics Workshop 2018: Analysis of Metagenomic Data](https://github.com/bioinformaticsdotca/Metagenomics_2018/blob/master/main_metagenomics.md) - Toronto June 5-7, 2018  
     
    * [Video lecture on shotgun metagenomics data analysis](https://www.youtube.com/watch?v=O3Gek4BkR-A&list=PL3izGL6oi0S_mH1C79CQNjVgCZy56WFoq&index=4) (YouTube link)   
    
    * [Shotgun metagenomics tutorial](https://github.com/LangilleLab/microbiome_helper/wiki/CBW-2018-Metagenomic-Taxonomic-and-Functional-Composition-Tutorial) (link to Microbiome Helper)
   
* Workflows for processing microbial amplicon sequencing data (all external links):
    * [Workflow dependent largely on QIIME 2](https://github.com/LangilleLab/microbiome_helper/wiki/Amplicon-SOP-v2-(qiime2-2020.2)) - Recommended pipeline (check Microbiome Helper for latest version)  

    * [Workflow for comparing the outputs of different taxa classifiers](https://github.com/gavinmdouglas/taxa_sanity_check)

    * [DADA2 custom wrapper script workflow](https://github.com/LangilleLab/microbiome_helper/wiki/DADA2-16S-Chemerin-Tutorial) - July 2018 

    * [QIIME 1 custom workflow](https://github.com/LangilleLab/microbiome_helper/wiki/16S-tutorial-(chemerin)) - October 2016  
  
