---
layout: page
title: Selected Publications
---

Below are four of my recent and representative publications. You can find my complete publication list at my [Google Scholar profile](https://scholar.google.ca/citations?hl=en&user=EhhXPUkAAAAJ).

**Douglas GM, Bielawski JP, Langille MGI. 2020. Re-evaluating the relationship between missing heritability and the microbiome. Microbiome 8:87.**
_A commentary focused on the issue of whether microbiome data can help address the issue of 'missing heritability'. In contrast to several recent articles, we argue that it would be inappropriate to consider microbial genetic variation in the human microbiome as an extension of the human genome. We then describe several approaches by which microbiome data could be used (given further evaluation) as a proxy for environmental variation in genome-wide association studies._


**Douglas GM, Maffei VJ, Zaneveld J, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. 2020. PICRUSt2 for prediction of metagenome functions. Nature Biotechnology 38:685-688.**
_An updated method for metagenome prediction based on amplicon sequences. We describe the updated steps in PICRUSt2 and also several new validation approaches that give an improved picture of how well metagenome prediction approaches perform. These improved validation approaches include computing the concordance expected by chance between predicted and actual metagenomes and also evaluating how similar the results of differential abundance testing are between the actual and predicted metagenome profiles._


**Douglas GM and Langille MGI. 2019. Current and Promising Approaches to Identify Horizontal Gene Transfer Events in Metagenomes. Genome Biology and Evolution 11:2750-2766.**
_Review paper describing the diverse set of approaches that have been developped for identifying horizontal gene transfer events in metagenomes and in isolate genomes. We contrast these approaches and describe how they have been applied. We close by describing several new technological advances that will likely greatly improve the performance of most bioinformatics approaches that we describe._


**Douglas GM\*, Hansen R\*, Jones C, Dunn K, Comeau AM, Bielawski JP, Tayler R, El-Omar EM, Russell RK, Hold GL, Langille MGI, Van Limbergen J. 2018. Multi-omics Differentially Classify Disease State and Treatment Outcome in Pediatric Crohn's Disease. Microbiome 6:13 (\* joint first authors).**
_A comparison of 16S rRNA gene sequencing and shotgun metagenomics sequencing of intestinal biopsy samples from pediatric Crohn's disease patients and healthy colon control patients. We assessed the performance of each datatype, collapsed to each taxonomic level and with functional annotations, for classifying patients by disease state and also by their response to treatment. Interestingly, the genera relative abundance values performed best for classifying patients by each category. We then created combined classifiers based on the top ranked features from each indivdual's datatype's model to assess which features perform best for classification overall. Although this analysis approach was useful it was noteworthy that the top features performed poorly for classification in a similar cohort. This result highlights that generalizing findings from clinical microbiome sequencing analyses across studies is challenging due to study-specific factors._
